dojo/importers/importer/importer.py

```diff
@@ -96,6 +96,24 @@ def process_parsed_findings(self, test, parsed_findings, scan_type, user, active
                 item.active = active
             if item.verified:
                 item.verified = verified
+            # Set the scan_date at import time to all findings imported
+            # Only allow this to happen if the scan_date is different than the default "today"
+            #   and different than what the scan report has.
+            # scan_date is set to both target_start and target_end in the creat_test function
+            target_start_date = test.target_start.date()
+            now_date = now.date()
+            if item.date == now_date:
+                # Parser did not set the date, so it is the default value
+                if target_start_date != now_date:
+                    # Import scan_date was set, and the parser has not overwritten it
+                    logger.debug('Parser did not set date, import override it ' + str(target_start_date) + ' : ' + str(now_date))
+                    item.date = test.target_start
+            else:
+                # The parser has set the date already
+                if target_start_date != now_date:
+                    # The date set by import scan_date should overwrite scan report date
+                    logger.debug('Parser set date, import override it ' + str(target_start_date) + ' : ' + str(now_date))
+                    item.date = test.target_start
```

Comment:
the if and else parts seem to do the identical thing, except logging a different message?

The PR description says:

_This PR sets the date of all findings imported to the date set at import time, but only if the parser fails to do so._

But the code seems to overwrite the date set by the parser in all cases? Looks like the unit tests check for that as well?

Not sure about how others use the scan_date, but I would expect that a date set by the parser will always be respected and not overwritten. So the scan_date is only used if the parser doesn't set a date, like the PR description mentions.

Reply:
I will change the PR description to be more descriptive, but no, the date is not overwritten in **all** cases. This is outlined in the unit tests created in this PR.

The motivation for this change is due to a scan being completed on given date (say 01/01/2005) and then imported to dojo at a later date. Regardless of setting the scan_date variable, the date of the findings will not change from those imported in the scan report. If the scan report does not specify the date, the date of the finding will always be the default date (the time of import) regardless of what value is specified in the scan_date. 

The default date given to a test (and all findings) is the date of import. This is true in a number of cases
 - the scan_date field is not supplied at all (API)
 - The scan_date is auto populated, and the user does not change it (UI)

The only way for a finding to have a date other than the date contained in the scan report (if the scanner even supplies it) or the default date (generated at import time)  is to manually set the date for every single finding imported.